spec.md

```diff
@@ -1492,6 +1502,187 @@ Applying integer decoding with a divisor of `100` to create an array of 32-bit f
 [ 1.00, 1.00, 1.00, 1.00, 0.50, 0.50 ]
 ```
 
+### Extra data
+
+The following are fields that are __not__ supplied by the *RCSB*, and are provided for applications to transfer non-molecular data along with the molecular data.
+
+Each field is a [map](#types), and there are 6 possible fields `bondProperties`, `atomProperties`, `groupProperties`, `chainProperties`,
+`modelProperties`, and `extraProperties`.  The first 5 fields are restricted to holding data that is relevant to the field's prefix, while
+any other data can be stored in the `extraProperties` field.  
+
+As mentioned in [Types](#types) in some fields we restrict the key-value types to specific types. The current type format the sepc follows is:
+* <String, Array|Binary>
+  * `atomProperties`
+  * `bondProperties`
+  * `groupProperties`
+  * `chainProperties`
+  * `modelProperties`
+* <String, String|Float|Integer|Map|Array|Binary>
+  * `extraProperties`
+
+For each of the fields above, we do reserve some key-value pair formats to help ensure cross-application-compatibility.  If you feel
+a certain key-value pair should be reserved and made a part of the spec, please let us know!  
+
+We encourage you to apply our encoding techniques to your application data to reduce file sizes!
+
+The restrictions on each of the lengths of the values in these fields are __not__ enforced at decode-time but are reserved to help guide applications.
+
+| Name            | length-restrictions     | 
+|-----------------|-------------------------|
+| bondProperties  | length of numBonds      |
+| atomProperties  | length of numAtoms      |
+| groupProperties | length of numGroups     |
+| chainProperties | length of numChains     |
+| modelProperties | length of numModels     |
+| extraProperties | None                    |
+
+Example
+
+```
+data = {
+  "mmtfVersion": "1.1",
+  "numAtoms": 999,
+  "numModels": 2,
+  "numChains": 4,
+  ...
+  "xCoordList": [1.2, 3.4, ...],
+  "yCoordList": [5.6, 7.8, ...],
+  "zCoordList": [9.0, 1.2, ...],
+  ...
+  "structureProperties": {
+    "foo_id": "ABC",
+  },
```

Comment:
There is an inconsistency with the specs here. We don't have a `structureProperties` field. I guess that properties at the structure level should go into `extraProperties`?

At least, I don't see the use case for having a `structureProperties` field vs adding structure-level properties in `extraProperties`.

Reply:
oops good point. Yes I'm ok with just wrapping it all into `extraProperties`. it would be sort of a useless field.